### DISCUSSION TITLE
Tap to click on wayland.

### DISCUSSION THREAD

**Discussion by ahmadraniri1994** [qtile/qtile#3583] [Q&A] [answered]
I'm sorry if this question was asked before. How to enable tap to click when use qtile Wayland. On X11, it seems get enabled by default. Are there some configurations I need to put on. Thanks.

**Comment by m-col** [comment, marked as answer]
Check out here: http://docs.qtile.org/en/stable/manual/wayland.html#inputconfig

You'll need to create an `InputConfig` to match your trackpad and sets `tap` to `True`. e.g.:
```py
wl_input_rules = {
    "type:pointer": InputConfig(tap=True),
}
```

**Reply by ahmadraniri1994** [reply]
I tried that but still no luck, how do we get the device's name ? In sway we can pass it with "*" so we don't bother to figure it out.

**Reply by m-col** [reply]
You can also pass `"*"` here.

**Reply by ahmadraniri1994** [reply]
Well, I'll check it again, the last time I tried I still can't do "tap to click" tough I did configure it. Thanks.

**Reply by m-col** [reply]
There is a bug with setting keyboard configuration, and while keyboards and pointers are configured differently, they both use InputConfig: https://github.com/qtile/qtile/issues/3447. It's possible there is a bug that's preventing yours from working, when it should.

Do you see any log output relating to your inputs when qtile starts up? I think currently `reload_config` doesn't apply new input configuration - this will change in the future.

**Reply by ahmadraniri1994** [reply]
> Do you see any log output relating to your inputs when qtile starts up?

No, I didn't see it.

I put this on my configuration : 
```
wl_input_rules = {
   "*": InputConfig(tap=True),
}
```
When I check it with `qtile check`, I get this : 

```
checking qtile config file /home/ahmad/.config/qtile/config.py
/tmp/tmpra4wrbk7/config.py:193: error: Argument "tap" to "InputConfig" has incompatible type "bool"; expected "Dict[str, Any]"
Found 1 error in 1 file (checked 1 source file)
config file type checking failed: Command '['mypy', '/tmp/tmpra4wrbk7/config.py']' 
returned non-zero exit status 1.
```

**Reply by m-col** [reply]
Your config looks fine. That `qtile check` output reveals a bug in the code - `bool` here is correct, it's the type annotation that is wrong.

Please could you start Qtile with `qtile start -l DEBUG` to get debug logs from the very beginning of startup, and post what you get? You may need to remove a lot of logs from after startup that aren't relevant.

**Reply by ahmadraniri1994** [reply]
This is what I get when I ran `qtile start -b wayland -l DEBUG` : 
[qtile.log](https://raw.githubusercontent.com/ahmadraniri1994/Semua-tentang-log/main/qtile_log)

**Reply by m-col** [reply]
Perfect, a lead!

The line:
```
AttributeError: 'NoneType' object has no attribute 'libinput_device_config_tap_get_finger_count'
```
indicates that your installation was not built with libinput support. The error message here is not very helpful, but this is fixed in master and so in the next release it will explicitly log that this is what has happened.

How did you install Qtile?

**Reply by ahmadraniri1994** [reply]
I use pip to install it,  instead of build from source. Man... thanks, I will try to build from source.

**Reply by ahmadraniri1994** [reply]
Solved by compile it from source. Thanks @m-col , I'm sorry to waste your time. I'll close it now.

**Reply by m-col** [reply]
Really it should work with pip. I'm not sure why it didnt build before. You shouldnt need source :/
